x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -127,32 +127,8 @@ private Object unwrapMultiValue(Object values) {
         if (values == null) {
             return null;
         }
-        if (dataType == DataType.GEO_POINT) {
-            Object value;
-            if (values instanceof List && ((List<?>) values).size() == 1) {
-                    value = ((List<?>) values).get(0);
-            } else {
-                value = values;
-            }
-            try {
-                GeoPoint geoPoint = GeoUtils.parseGeoPoint(value, true);
-                return new GeoShape(geoPoint.lon(), geoPoint.lat());
-            } catch (ElasticsearchParseException ex) {
-                throw new SqlIllegalArgumentException("Cannot parse geo_point value (returned by [{}])", fieldName);
-            }
-        }
-        if (dataType == DataType.GEO_SHAPE) {
-            Object value;
-            if (values instanceof List && ((List<?>) values).size() == 1) {
-                value = ((List<?>) values).get(0);
-            } else {
-                value = values;
-            }
-            try {
-                return new GeoShape(value);
-            } catch (IOException ex) {
-                throw new SqlIllegalArgumentException("Cannot read geo_shape value (returned by [{}])", fieldName);
-            }
+        if (dataType == DataType.GEO_POINT || dataType == DataType.GEO_SHAPE) {
```

Comment:
I think it's worth adding an `isGeoBased()` method to `DataType` to perform this double check.